### PR TITLE
Fix a bug where the diagnostics file dropped all conversion problems

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://115521475

## Summary

This fixes a bug where the DiagnosticsFileWriter dropped all the problems from the documentation conversion because `flush()` is called twice and the file writer assumed that it would only be called once and removed all the gathered problem after writing them to a file. This resulted in the next `flush()` call to write an empty diagnostics file.

## Dependencies

None

## Testing

- Build documentation for some project with documentation warnings (for example unresolved symbol links) and pass \
   `--diagnostics-file /path/to/something.json`

- Inspect the resulting diagnostics json file. It should contain entries for all diagnostics from the build.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticFileWriter.swift [hunk 1]
```diff
@@ -29,10 +29,15 @@ public final class DiagnosticFileWriter: DiagnosticConsumer {
...
-        receivedProblems = []
+        // Don't clear the received problems, `flush()` is called more than once.
         let encoder = RenderJSONEncoder.makeEncoder(emitVariantOverrides: false)
         try encoder.encode(fileContent).write(to: outputPath, options: .atomic)
     }
+    
+    // This is deprecated but still necessary to implement.
+    public func finalize() throws {
+        try flush()
+    }
```
Comment (sofiaromorales) [inline Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticFileWriter.swift:42]:
```suggestion
    @available(*, deprecated, renamed: "flush()")
    public func finalize() throws {
        try flush()
    }
```

**Review by sofiaromorales — APPROVED** [review]
Besides the minor comment this looks good to me 👍

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**